beta-src/src/models/PathClass.ts

```diff
@@ -0,0 +1,337 @@
+
+          // now search alternativeRoute for alternativeRoutes (beginning at the node next to end node)
+          alternativeRoute.pathToNode.searchAlternativeRoutes(fEndNode);
+          /*
+           * Note, that not extra check is needed for pathToNode:
+           * Even if the alternativeRoute heads directly into endNode,
+           * pathToNode is at least this. So the extreme case is that other
+           * paths from this node are checked (which is wanted)
+           */
+        } else {
+          // all alternative routes from this node are found
+          this.alternativeChecked = true;
+        }
+      }
+    }
+
+    if (this.pathToNode !== null)
+      this.pathToNode.searchAlternativeRoutes(fEndNode);
+  }
+
+  /*
+   * Returns an alternative path that does not pass through this.node.
+   * Alternative routes have to be searched before-
+   */
+  getAlternativeRoute() {
+    if (!this.alternativeChecked) return null;
+
+    if (this.pathToNode === null) return null;
+
+    if (this.pathToNode.pathNextNodes.length > 1) {
+      // there is a fork in the path at node before
+      // find a path to next node of previous node that is not this.
+      // and return the complete path
+      return this.pathToNode.pathNextNodes
+        .find((path) => {
+          return path !== this;
+        }, this)
+        .getLastPathNode();
+    }
+    return this.pathToNode.getAlternativeRoute();
+  }
+
+  /*
+   * Checks if this path can be appended to path so a simple path is preserved.
+   * So this basically checks if both paths share any common nodes apart from
+   * path.lastNode and this.firstNode.
+   *
+   * Implemented recursivly
+   */
+  canBeAppendedTo(path) {
+    path.markPath(path);
+
+    let retValue = true;
+
+    if (this.node.inPath === path) retValue = false;
+    else if (this.pathToNode !== null)
+      retValue = this.pathToNode.canBeAppendedTo(path);
+
+    return retValue;
+  }
+
+  markPath(path) {
+    if (this.node.inPath === path) return;
+
+    this.node.inPath = path;
+
+    if (this.pathToNode !== null) this.pathToNode.markPath(path);
+  }
+
+  toArray(array: string[] = []) {
+    // if (Object.isUndefined(array)) array = new Array();
+
+    // do not include last element of path in array representation
+    array.push(this.node.id);
+
+    if (this.pathToNode != null) {
+      return this.pathToNode.toArray(array);
+    }
+
+    return array;
+  }
+
+  getLength() {
+    return 1 + (this.pathToNode !== null ? this.pathToNode.getLength() : 0);
+  }
+
+  // DEBUGGING ONLY
+  getFirst() {
+    if (this.pathToNode === null) {
+      return this;
+    }
+
+    return this.pathToNode.getFirst();
+  }
+
+  getAllPathsFromThis() {
+    if (this.pathNextNodes.length === 0) {
+      return this;
+    }
+    return this.pathNextNodes
+      .map((pnn) => {
+        return pnn.getAllPathsFromThis();
+      })
+      .flat();
+    // return this.pathNextNodes.invoke("getAllPathsFromThis").flatten();
```

Comment:
remove all commented code. if you need to retain it for some reason, make a comment that describes what used to be present and why it was commented, etc. usually we just removed commented code